--- rrd_restore_orig.c  2009-05-19 23:45:05.000000000 +1000
+++ rrd_restore.c       2009-07-04 10:52:02.984375000 +1000
@@ -48,6 +48,7 @@
 #define ARRAY_LENGTH(a) (sizeof (a) / sizeof ((a)[0]))
 static int opt_range_check = 0;
 static int opt_force_overwrite = 0;
+long * timet2long;

 /*
  * Auxiliary functions
@@ -91,10 +92,7 @@
     return (0);
 }                       /* int get_string_from_node */

-static int get_long_from_node(
-    xmlDoc * doc,
-    xmlNode * node,
-    long *value)
+static int get_long_from_node( xmlDoc * doc, xmlNode * node, long *value)
 {
     long       temp;
     char     *str_ptr;
@@ -915,11 +913,12 @@
                                           rrd->stat_head->version,
                                           sizeof(rrd->stat_head->version));
         else if (xmlStrcmp(child->name, (const xmlChar *) "step") == 0)
-            status = get_ulong_from_node(doc, child,
-                                        &rrd->stat_head->pdp_step);
+            status = get_ulong_from_node(doc, child,&rrd->stat_head->pdp_step);
         else if (xmlStrcmp(child->name, (const xmlChar *) "lastupdate") == 0)
-            status = get_long_from_node(doc, child,
-                                        &rrd->live_head->last_up);
+               {
+                       timet2long = (long *) &rrd->live_head->last_up;
+            status = get_long_from_node(doc, child,timet2long);
+               }
         else if (xmlStrcmp(child->name, (const xmlChar *) "ds") == 0)
             status = parse_tag_ds(doc, child, rrd);
         else if (xmlStrcmp(child->name, (const xmlChar *) "rra") == 0)
